src/App.js

```diff
@@ -65,12 +65,26 @@ export class ResourceEditor extends React.Component {
     }
   }
 
-  metadataHandler(resource) {
+  metadataHandler(fileResource) {
+    let resource = this.extendResource(fileResource)
     this.setState({
       resource,
     });
   }
 
+  /**
+   * Adds a new field bq_table_name to resources and formats name property
+   * @param {Object} resource 
+   */
+  extendResource(resource){
+    let newResource = {...resource}
+    newResource.title = resource.name
+    newResource.name = resource.name
+
+    let new_name = resource.name.split(".")[0].toUpperCase()
```

Comment:
@risenW could be good to change `new_name` to `newName` looks like python :)

 and just a best practice in JS if you do not change the variable you can protect using `const` stead `let`. So in the future of someone trying to change the new_name variable will return an error.

Reply:
Sure, thanks. Will update now